sdk/cosmos/azure_data_cosmos/examples/cosmos/main.rs

```diff
@@ -74,11 +74,11 @@ pub async fn main() -> Result<(), Box<dyn Error>> {
     }
 }
 
-fn create_client(args: &SharedArgs) -> Result<CosmosClient, Box<dyn Error>> {
-    if let Some(key) = args.key.as_ref() {
+fn create_client(args: &mut SharedArgs) -> Result<CosmosClient, Box<dyn Error>> {
+    if let Some(key) = args.key.take() {
```

Comment:
Couldn't this make the API harder to use? Is there any reason to `take()`?

Reply:
This is the only time the sample needs to read this argument, but see what you're saying. There's not really a specific _reason_ other than I kinda wanted to consume the key here. I can swap it back.

Reply:
Oh, I remember. There's no `impl From<&String> for Secret`. So I have to clone, which feels kinda unnecessary. But I'm optimizing a sample that's _really_ just designed for testing, so I should probably stop worrying about it 😅

Reply:
(To be fair, I was cloning before too)